chain/src/chain.rs

```diff
@@ -703,9 +699,25 @@ impl BlockChain {
             &chain_id,
             None,
         )?;
+        dag = Self::init_dag(dag, genesis_header)?;
         Self::new(time_service, executed_block.block.id(), storage, None, dag)
     }
 
+    fn init_dag(mut dag: BlockDAG, genesis_header: BlockHeader) -> Result<BlockDAG> {
```

Comment:
Is it reasonable if a newly created chain has  `dag_state`? Genesis block is only executed when creating a chain. For non-empty chain, it will be loaded and checked.

Reply:
Dag state stores the tips or other related data though it only stores tips and genesis id for now.

Reply:
For a new chain, this case should be treat as dirty data.